File: 김하영/two pointer/3273.js
const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const n = +input[0];

const arr = input[1].split(" ").map((v) => +v);
const x = +input[2];

const twoPointer = (arr, x) => {
  arr.sort((a, b) => a - b);
  let answer = 0;
  let start = 0;
  let end = arr.length - 1;

  while (start < end) {
    if (arr[start] + arr[end] === x) {
      answer++;
      start++;
    } else if (arr[start] + arr[end] > x) {
      end--;
    } else {
      start++;
    }
  }

  return answer;
};

console.log(twoPointer(arr, x));

File: BoSeonPark/투 포인터/3273.py
import sys
input = sys.stdin.readline

n = int(input())
a = list(map(int, input().split()))
x = int(input())
a.sort()
l, r = 0, n - 1
answer = 0

while l != r:
    if a[l] + a[r] == x:
        answer += 1
        r -= 1
    elif a[l] + a[r] < x:
        l += 1
    elif a[l] + a[r] > x:
        r -= 1

print(answer)